File: src/main/java/cz/forgottenempire/arma3servergui/workshop/ModDto.java
package cz.forgottenempire.arma3servergui.workshop;

import cz.forgottenempire.arma3servergui.common.ServerType;
import javax.persistence.Id;
import lombok.Data;

@Data
public class ModDto {

    @Id
    private Long id;
    private String name;
    private ServerType serverType;
    private Long fileSize;
    private String lastUpdated;
    private String installationStatus;
    private String errorStatus;
}

File: src/main/webapp/src/dtos/ModDto.ts
import {ServerType} from "./ServerDto.ts";
import {ErrorStatus, InstallationStatus} from "./Status.ts";

export interface ModDto {
    id: number,
    name: string,
    serverType: ServerType,
    fileSize: number,
    lastUpdated: string,
    installationStatus: InstallationStatus,
    errorStatus: ErrorStatus
}
